packages/design-system/src/components/web-components/ds-modal-dialog/ds-modal-dialog.stories.tsx

```diff
@@ -0,0 +1,268 @@
+  args: {
+    alert: 'false',
+    children: (
+      <div>
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed accumsan diam vitae metus
+        lacinia, eget tempor purus placerat.
+      </div>
+    ),
+    heading: 'Dialog Heading',
+    'is-open': 'false',
+  },
+  argTypes: {
+    'actions-class-name': {
+      description: 'Additional classes to be added to the actions container.',
+      control: 'text',
+    },
+    alert: {
+      description:
+        'If `true`, the modal will receive a role of `alertdialog`, instead of its default `dialog`. The `alertdialog` role should only be used when an alert, error, or warning occurs.',
+      control: 'boolean',
+    },
+    'backdrop-click-exits': {
+      description:
+        'Pass `true` to have the dialog close when its backdrop pseudo-element is clicked',
+      control: 'boolean',
+    },
+    children: {
+      table: {
+        disable: true,
+      },
+    },
+    'class-name': {
+      description: 'Additional classes to be added to the root dialog element.',
+      control: 'text',
+    },
+    'dialog-close-label': {
+      description: 'Aria label for the close button',
+      control: 'text',
+      table: {
+        defaultValue: { summary: 'Close modal dialog' },
+      },
+    },
+    'header-class-name': {
+      description:
+        'Additional classes to be added to the header, which wraps the heading and close button.',
+      control: 'text',
+    },
+    heading: {
+      description: "The dialog's heading, to be rendered in the header alongside the close button.",
+      control: 'text',
+    },
+    'root-id': {
+      description: 'A custom `id` attribute for the dialog element',
+      control: 'text',
+    },
+    'is-open': {
+      description: 'Controls whether the dialog is in an open state',
+      control: 'boolean',
+    },
+    size: {
+      description: 'Can be one of three values: narrow, wide and full',
+      options: ['narrow', 'wide', 'full'],
+      control: 'radio',
+    },
+  },
+  parameters: {
+    docs: {
+      page: WebComponentDocTemplate,
+      description: {
+        component:
+          'For information about how and when to use this component, [refer to its full documentation page](https://design.cms.gov/components/dialog/).',
+      },
+      componentEvents: {
+        'ds-exit': {
+          description: 'A callback function triggered when the user clicks the filter chip button.',
+        },
+      },
+      sharedAttrLists: ['analytics'],
+      slots: {
+        actions: {
+          description:
+            'Buttons or other HTML to be rendered in the "actions" bar at the bottom of the dialog.',
+        },
+        heading: {
+          description:
+            "The dialog's heading, to be rendered in the header alongside the close button.",
+        },
+      },
+    },
+  },
+  decorators: [webComponentDecorator],
+};
+export default meta;
+
+const LongContent = ({ args }) => {
+  useEffect(() => {
```

Comment:
Is there a more concise way of doing this? I tried creating the LongContent section as it's own component in this file and then passing in the Template content as children props. This didn't seem to work. I'd prefer to not have to duplicate this useEffect if possible.

Reply:
Since `Template` is just a component function, could we render it like a component inside your `PreventScrollExample` story like this?

```ts
export const PreventScrollExample = {
  render: function Component(args) {
    return (
      <div>
        all the long content before the template
        <Template {...args} />
        all the long content after the button
      </div>
    );
  },
```

Reply:
You could put this `useEffect` logic into its own decorator:

```
const ModalDecorator = (Story: React.FC) => {
  useEffect(() => {
    const button = document.querySelector('ds-button');
    const modal = document.querySelector('ds-modal-dialog');
    const toggleModal = (event): void => {
      const targetType = event.target.nodeName ?? '';
      if (targetType === 'DS-BUTTON') {
        action('ds-click')(event);
      } else {
        action('ds-exit')(event);
      }
      const isOpen = modal.getAttribute('is-open');
      if (isOpen === 'true') {
        modal.setAttribute('is-open', 'false');
      } else {
        modal.setAttribute('is-open', 'true');
      }
    };
    button.addEventListener('ds-click', toggleModal);
    modal.addEventListener('ds-exit', toggleModal);
    return () => {
      button.removeEventListener('ds-click', toggleModal);
      modal.removeEventListener('ds-exit', toggleModal);
    };
  });

  return (
    <Story />
  )
}
```

You would add the following to each story object `decorators: [ModalDecorator]`